### PR TITLE
Add builtin ArbStatistics contract making available chain statistics

### PR DESCRIPTION
**Pull request by edfelten**

### CODE REVIEW

File: arb_os/evmCallStack.mini [hunk 1]
```diff
@@ -783,6 +785,8 @@ public impure func evmCallStack_returnFromCall(
                             Some(gasUsage),
                             feeStats,
                         );
+
+                        arbStatistics_addContract();
```
Comment (hkalodner) [inline arb_os/evmCallStack.mini:789]: This covers externally created contracts, but does it cover `CREATE` and `CREATE2`?

File: arb_os/arbstatistics.mini [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    } else {
+        evmOp_revert_knownPc(12, 0, 0);
+    }
+}
+
+var arbStats: struct {
+    storage: uint,
+    arbGasUsed: uint,
+    numTxReceipts: uint,
+    numContracts: uint,
```
Comment (hkalodner) [inline arb_os/arbstatistics.mini:69]: It looks like this doesn't take get reduced by `SELFDESTRUCT`. Is that intentional? If so maybe `numContractsCreated` would be better

**Review by hkalodner — CHANGES_REQUESTED** [review]

File: arb_os/evmCallStack.mini (hunk 1 above)
Comment (edfelten) [reply arb_os/evmCallStack.mini]: Added coverage for CREATE and CREATE2

File: arb_os/arbstatistics.mini (hunk 2 above)
Comment (edfelten) [reply arb_os/arbstatistics.mini]: Added coverage for SELFDESTRUCT

**Review by hkalodner — APPROVED** [review]
LGTM

**Pull request merged**